File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {
        int fnt=0,bk=0,l=s.length();
        char[] ans=s.toCharArray();
        s="";
        while(fnt!=l){
            if(ans[fnt]==32){
                int temp=fnt-1;
                while(temp>=bk){
                    s+=ans[temp];
                    temp--;
                }
                s+=" ";
                bk=fnt+1;
            }else if(fnt==l-1){
                int temp=fnt;
                while(temp>=bk){
                    s+=ans[temp];
                    temp--;
                }
            }
            fnt++;
        }
        return s;
    }
}

File: 557-reverse-words-in-a-string-iii/557-reverse-words-in-a-string-iii.py
class Solution:
    def reverseWords(self, s: str) -> str:
        arr=s.split(" ")
        for i in range(len(arr)):
            j, k = "",len(arr[i])-1
            while k>=0:
                j+=arr[i][k]
                k-=1
            arr[i]=j
        ans=" "
        return ans.join(arr)
        